cmdstanpy/cmdstan_args.py

```diff
@@ -0,0 +1,398 @@
+        return cmd
+
+
+class FixedParamArgs(object):
+    """Arguments for the NUTS adaptive sampler."""
+
+    def compose(self, idx: int, cmd: str) -> str:
+        cmd = cmd + ' method=fixed_param'
+        return cmd
+
+    def validate(self, chains):
+        pass
+
+
+class CmdStanArgs(object):
+    """
+    Container for CmdStan command line arguments.
+    Consists of arguments common to all methods and
+    and an object which contains the method-specific arguments.
+    """
+
+    def __init__(
+        self,
+        model_name: str,
+        model_exe: str,
+        chain_ids: Union[List[int], None],
+        method_args: Union[SamplerArgs, FixedParamArgs],
+        data: str = None,
+        seed: Union[int, List[int]] = None,
+        inits: Union[float, str, List[str]] = None,
+        output_basename: str = None,
+    ) -> None:
+        """Initialize object."""
+        self.model_name = model_name
+        self.model_exe = model_exe
+        self.chain_ids = chain_ids
+        self.method_args = method_args
+        self.method_args.validate(len(chain_ids) if chain_ids else None)
+        self.data = data
+        self.seed = seed
+        self.inits = inits
+        self.output_basename = output_basename
+        self.validate()
+
+    def validate(self) -> None:
+        """
+        Check arguments correctness and consistency.
+
+        * input files must exist
+        * output files must be in a writeable directory
+        * if no seed specified, set random seed.
+        * length of per-chain lists equals specified # of chains
+        """
+        if self.model_name is None:
+            raise ValueError('no stan model specified')
+        if self.model_exe is None:
+            raise ValueError('model not compiled')
+
+        if self.chain_ids is not None:
+            for i in range(len(self.chain_ids)):
+                if self.chain_ids[i] < 1:
+                    raise ValueError(
+                        'invalid chain_id {}'.format(self.chain_ids[i])
+                    )
+
+        if self.output_basename is not None:
+            if not os.path.exists(os.path.dirname(self.output_basename)):
+                raise ValueError(
+                    'invalid path for output files: {}'.format(
+                        self.output_basename)
+                )
+            try:
+                with open(self.output_basename, 'w+') as fd:
+                    pass
+                os.remove(self.output_basename)  # cleanup
+            except Exception:
+                raise ValueError(
+                    'invalid path for output files: {}'.format(
+                        self.output_basename)
+                )
+            if self.output_basename.endswith('.csv'):
+                self.output_basename = self.output_basename[:-4]
+
+        if self.seed is None:
+            rng = np.random.RandomState()
+            self.seed = rng.randint(1, 99999 + 1)
```

Comment:
Is this the correct maximum?

Reply:
yes - see https://docs.scipy.org/doc/numpy/reference/generated/numpy.random.randint.html
 >high : int, optional
> If provided, one above the largest (signed) integer to be drawn from the distribution